core/error/error_list.cpp

```diff
@@ -31,55 +31,55 @@
 #include "error_list.h"
 
 const char *error_names[] = {
-	"No error",
-	"Generic error",
-	"Requested operation is unsupported/unavailable",
-	"The object hasn't been set up properly",
-	"Missing credentials for requested resource",
-	"Parameter out of range",
-	"Out of memory",
-	"File not found",
-	"Bad drive",
-	"Bad path",
-	"Permission denied",
-	"Already in use",
-	"Can't open file",
-	"Can't write file",
-	"Can't read file",
-	"File unrecognized",
-	"File corrupt",
-	"Missing dependencies for file",
-	"Unexpected eof",
-	"Can't open resource/socket/file", // File too? What's the difference to ERR_FILE_CANT_OPEN
-	"Can't create", // What can't be created,
-	"Query failed", // What query,
-	"Already in use",
-	"Resource is locked",
-	"Timeout",
-	"Can't connect",
-	"Can't resolve hostname", // I guessed it's the hostname here.
-	"Connection error",
-	"Can't acquire resource",
-	"Can't fork",
-	"Invalid data",
-	"Invalid parameter",
-	"Item already exists",
-	"Item does not exist",
-	"Can't read from database", // Comments say, it's full? Is that correct?
-	"Can't write to database", // Is the database always full when this is raised?
-	"Compilation failed",
-	"Method not found",
-	"Link failed",
-	"Script failed",
-	"Cyclic link detected",
-	"Invalid declaration",
-	"Duplicate symbol",
-	"Parse error",
-	"Resource is busy",
-	"Skip error", // ???? What's this? String taken from the docs
-	"Help error", // More specific?
-	"Bug",
-	"Printer on fire",
+	"OK", // OK
+	"Failed", // FAILED
+	"Unavailable", // ERR_UNAVAILABLE
+	"Unconfigured", // ERR_UNCONFIGURED
+	"Unauthorized", // ERR_UNAUTHORIZED
+	"Parameter out of range", // ERR_PARAMETER_RANGE_ERROR
+	"Out of memory", // ERR_OUT_OF_MEMORY
+	"Not found", // ERR_FILE_NOT_FOUND
```

Comment:
If this is an example of "generalizes error messages so they work in more contexts", then I'm not sure... The constant is still `ERR_FILE_NOT_FOUND`, so we can't really use it anywhere unrelated to files. Same for the permission denied below (which you didn't touch, but it's similar).

Reply:
TBH, my plan is to move in the opposite direction.

We should have a general ERR_NOT_FOUND, not a special one that only some file methods return (because then this should be moved to a FileError enum).

Same for the permission one, that can be ERR_UNAUTHENTICATED.

I also tried to keep these short and general, because while having whole sentences with explanations can be nice in some cases, it will read real weird or outright wrong in some cases the errors are currently used/returned.

But thats another PR :)

Reply:
^ Resolved by adding mentions of "File" back, and generalizing error messages later when we also change the constants.